test/utils/crd/crd_util.go

```diff
@@ -69,6 +69,56 @@ func CreateMultiVersionTestCRD(f *framework.Framework, group string, opts ...Opt
 		return nil, err
 	}
 
+	crd := genRandomCRD(f, group, opts...)
+	// Be robust about making the crd creation call.
+	var got *apiextensionsv1.CustomResourceDefinition
+	if err := wait.PollUntilContextTimeout(context.TODO(), f.Timeouts.Poll, 30*time.Second, true, func(ctx context.Context) (bool, error) {
+		// Create CRD and waits for the resource to be recognized and available.
+		got, err = fixtures.CreateNewV1CustomResourceDefinitionWatchUnsafe(crd, apiExtensionClient)
+		if err != nil {
+			if apierrors.IsAlreadyExists(err) {
+				// regenerate on conflict
+				framework.Logf("CustomResourceDefinition name %q was already taken, generate a new name and retry", crd.Name)
+				crd = genRandomCRD(f, group, opts...)
+			} else {
+				framework.Logf("Unexpected error while creating CustomResourceDefinition: %v", err)
+			}
+			return false, nil
+		}
+		return true, nil
+	}); err != nil {
+		return nil, err
+	}
+
+	resourceClients := map[string]dynamic.ResourceInterface{}
+	for _, v := range got.Spec.Versions {
+		if v.Served {
+			gvr := schema.GroupVersionResource{Group: got.Spec.Group, Version: v.Name, Resource: got.Spec.Names.Plural}
+			resourceClients[v.Name] = dynamicClient.Resource(gvr).Namespace(f.Namespace.Name)
+		}
+	}
+
+	testcrd.APIExtensionClient = apiExtensionClient
+	testcrd.Crd = got
+	testcrd.DynamicClients = resourceClients
+	testcrd.CleanUp = func(ctx context.Context) error {
+		err := fixtures.DeleteV1CustomResourceDefinition(got, apiExtensionClient)
+		if err != nil {
+			framework.Failf("failed to delete CustomResourceDefinition(%s): %v", got.Name, err)
+		}
+		return err
+	}
+	return testcrd, nil
+}
+
+// genRandomCRD generates a random CRD name and kind based on the framework's base name and a random suffix.
+// It also sets the group to the provided value and sets the scope to NamespaceScoped. If no versions are provided via
+// the opts, it will create a default version "v1" with an allow-all schema.
+func genRandomCRD(f *framework.Framework, group string, opts ...Option) *apiextensionsv1.CustomResourceDefinition {
+	suffix := framework.RandomSuffix()
+	name := fmt.Sprintf("e2e-test-%s-%s-crd", f.BaseName, suffix)
```

Comment:
we have `UniqueName` just for this
```suggestion
	name := fmt.Sprintf("e2e-test-%s-%s-crd", f.UniqueName, suffix)
```